Products/CMFPlone/browser/search.py

```diff
@@ -183,10 +184,27 @@ def filter_types(self, types):
         plone_utils = getToolByName(self.context, "plone_utils")
         if not isinstance(types, list):
             types = [types]
-        return plone_utils.getUserFriendlyTypes(types)
+
+        # Get the vocabulary
+        vocab_factory = self._portal_types_vocabulary()
```

Comment:
I find the `_portal_types_vocabulary` method unnecessary.

Reply:
yes, just add `queryUtility` here - do not blow up blown up code more than needed.

Reply:
done